src/components/expired-subscription-modal/index.jsx

```diff
@@ -1,25 +1,52 @@
 import {
-  useToggle, AlertModal, Button, ActionRow,
+  useToggle, AlertModal, ActionRow, StatefulButton,
 } from '@openedx/paragon';
 import DOMPurify from 'dompurify';
-import { useSubscriptions } from '../app/data';
+import { useContext, useState } from 'react';
+import { AppContext } from '@edx/frontend-platform/react';
+import { postUnlinkUserFromEnterprise, useEnterpriseCustomer, useSubscriptions } from '../app/data';
 
 const ExpiredSubscriptionModal = () => {
+  const [buttonState, setButtonState] = useState('default');
   const { data: { customerAgreement } } = useSubscriptions();
+  const {
+    authenticatedUser: { email: userEmail },
+  } = useContext(AppContext);
+  const { data: enterpriseCustomer } = useEnterpriseCustomer();
+
   const [isOpen] = useToggle(true);
   if (!customerAgreement?.hasCustomLicenseExpirationMessaging) {
     return null;
   }
+  const onClickHandler = (redirectUrl) => async (e) => {
+    e.preventDefault();
+    setButtonState('pending');
+
+    await postUnlinkUserFromEnterprise(enterpriseCustomer.uuid, userEmail);
+
+    // Redirect immediately
+    window.location.href = redirectUrl;
+    setButtonState('default');
+  };
+  const props = {
+    labels: {
+      default: customerAgreement.buttonLabelInModal,
+      pending: 'Redirecting...',
+    },
+    variant: 'primary',
+  };
   return (
     <AlertModal
       title={<h3 className="mb-2">{customerAgreement.modalHeaderText}</h3>}
       isOpen={isOpen}
       isBlocking
       footerNode={(
         <ActionRow>
-          <Button href={customerAgreement.urlForButtonInModal}>
-            {customerAgreement.buttonLabelInModal}
-          </Button>
+          <StatefulButton
+            state={buttonState}
+            onClick={onClickHandler(customerAgreement.urlForButtonInModal)}
```

Comment:
I believe the `onClickHandler` is executed immediately when the component renders, not when the user clicks the button.

If you need to pass `customerAgreement.urlForButtonInModal` as a function arg, then `onClickHandler` should be wrapped in an anonymous function so it doesn't get called upon render.

```jsx
onClick={() => onClickHandler(customerAgreement.urlForButtonInModal)}
```

Alternatively, I don't think the `onClickHandler` function necessarily needs to accept the redirect URL as a function arg as it's accessible in scope, e.g.:

```jsx
const onClickHandler = () => async (e) => {
  e.preventDefault();
  setButtonState('pending');

  await postUnlinkUserFromEnterprise(enterpriseCustomer.uuid, userEmail);

  // Redirect immediately
  window.location.href = customerAgreement.urlForButtonInModal;
  setButtonState('default');
};

// Usage
onClick={onClickHandler}
```

Reply:
[Justification] Actually, it was not showing any API call in the network tab when the modal renders which is why I made myself forward on this approach.

Your provided second approach makes more sense.